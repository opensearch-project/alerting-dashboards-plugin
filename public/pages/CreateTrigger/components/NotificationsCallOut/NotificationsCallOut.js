/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';

const NotificationsCallOut = () => {
  return (
    <div>
      <EuiCallOut title="Notifications plugin is not installed" color="danger" iconType="alert">
        <p>
          Install the notifications plugin in order to create and select channels to send out
          notifications.{' '}
          <EuiLink href="#" external>
            Learn more
          </EuiLink>
          .
        </p>
      </EuiCallOut>
      <EuiSpacer size="m" />
    </div>
  );
};

export default NotificationsCallOut;
