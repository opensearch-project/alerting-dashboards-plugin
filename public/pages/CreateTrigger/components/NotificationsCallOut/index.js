/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import NotificationsCallOut from './NotificationsCallOut';

export default NotificationsCallOut;
