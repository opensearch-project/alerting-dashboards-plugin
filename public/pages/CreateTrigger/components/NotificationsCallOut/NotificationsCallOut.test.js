/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';

import NotificationsCallOut from './NotificationsCallOut';

describe('NotifictionsCallOut', () => {
  test('renders', () => {
    const component = <NotificationsCallOut />;

    expect(render(component)).toMatchSnapshot();
  });
});
