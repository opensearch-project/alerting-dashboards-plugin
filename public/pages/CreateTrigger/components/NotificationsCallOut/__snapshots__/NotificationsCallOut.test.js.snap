// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotifictionsCallOut renders 1`] = `
<div>
  <div
    class="euiCallOut euiCallOut--danger"
  >
    <div
      class="euiCallOutHeader"
    >
      <svg
        aria-hidden="true"
        class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiCallOutHeader__icon"
        focusable="false"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      />
      <span
        class="euiCallOutHeader__title"
      >
        Notifications plugin is not installed
      </span>
    </div>
    <div
      class="euiText euiText--small"
    >
      <div
        class="euiTextColor euiTextColor--default"
      >
        <p>
          Install the notifications plugin in order to create and select channels to send out notifications. 
          <a
            class="euiLink euiLink--primary"
            href="#"
            rel="noreferrer"
          >
            Learn more
            <svg
              aria-hidden="true"
              aria-label="External link"
              class="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            />
          </a>
          .
        </p>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--m"
  />
</div>
`;
